src/rubrix/server/daos/records.py

```diff
@@ -287,10 +303,19 @@ def scan_dataset(
         es_query = {
             "query": search.query or {"match_all": {}},
             "highlight": self.__configure_query_highlight__(task=dataset.task),
+            "sort": [{"id": {"order": "asc"}}]  # Sort the search so the consistency is maintained in every search
         }
-        docs = self._es.list_documents(
-            dataset_records_index(dataset.id), query=es_query
-        )
+        if id_from:
+            # Scroll method does not accept read_after, thus, this case is handled as a search
+            es_query["search_after"] = [id_from]
+            results = self._es.search(index=dataset_records_index(dataset.id), query=es_query, size=limit)
+            hits = results["hits"]
+            docs = hits["hits"]
+
+        else:
+            docs = self._es.list_documents(
+                dataset_records_index(dataset.id), query=es_query,
+            )
```

Comment:
Could we use the `search` method always for dataset loading? That means that the `rb.load` will fetch records always in batches, and can improve the client's overall performance. 

We avoid using index scrolling, what is a recommendation from the [official docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/scroll-api.html)

Reply:
Agree :D